### PR TITLE
Enhance hover support

### PR DESCRIPTION
**Pull request by H3mul**
A quick little adjustment I was missing: using hover preview to quickly go through notes in Board view.

I also took the opportunity adjust the hover treatment of Table view to a more preferred setup: instead of handling the meta keys in projects, we should register a new hover view source, that way the meta key treatment can be toggled inside Page Preview settings:

image

Tested and made sure this works with both core Page Preview and the Hover Editor plugin.

### CODE REVIEW

**Comment by Acylation** [comment]
Nice catch!

File: src/ui/views/Table/components/DataGrid/GridRow.svelte [hunk 1]
```diff
@@ -54,34 +53,9 @@
       }
     };
   }
-
-  function handleHoverLink(event: MouseEvent) {
-    if (!event.ctrlKey && !event.metaKey) {
-      return;
-    }
```
Comment (Acylation) [inline src/ui/views/Table/components/DataGrid/GridRow.svelte:61]: With regard to this condition, the `default mod` option should be true to get better backwards-compatibility.

Comment (H3mul) [reply src/ui/views/Table/components/DataGrid/GridRow.svelte:61]: Sounds good

**Comment by H3mul** [comment]
Since InternalLink is moving to obsidian-projects, perhaps take a look at my PR to obsidian-svelte:
https://github.com/marcusolsson/obsidian-svelte/pull/9

**Comment by Acylation** [comment]
Hi I just saw https://github.com/marcusolsson/obsidian-svelte/pull/9. Thanks for pointing out the problem and the fix! Since we gonna use a mocked `InternalLink` component in this PR ([59d87b6](https://github.com/marcusolsson/obsidian-projects/pull/898/commits/59d87b6178f066736d3e84cfdd07a0b5b5003fea)), here I also include your fix for a temporary solution. I will contribute this component back after some refactoring (decouple dom-event handler and the hover-trigger).

Closes #780.

**Pull request merged**